### PR TITLE
add gitlab collector

### PR DESCRIPTION
**Pull request by benathmane**

### CODE REVIEW

File: collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/DefaultGitlabGitClient.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    private static final String PUBLIC_GITLAB_HOST_NAME = "gitlab.company.com";
+
+    @Autowired
+    public DefaultGitlabGitClient(GitlabSettings gitlabSettings,
+                                       Supplier<RestOperations> restOperationsSupplier) {
+        this.gitlabSettings = gitlabSettings;
+        this.restOperations = restOperationsSupplier.get();
+    }
+
+    private TrustManager[ ] get_trust_mgr() {
```
Comment (MarkRx) [inline collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/DefaultGitlabGitClient.java]: You probably need to add a root certificate to your java store. Trusting arbitrary certs is not a safe solution.

Comment (benathmane) [inline collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/DefaultGitlabGitClient.java]: But it works in my local :)

capture

**Comment by tabladrum** [comment]
@benathmane can you see why the build failed?

**Comment by amitmawkin** [comment]
@tabladrum the build succeeded, do you have any other concern before we begin our merging activities and introduce this collector.

**Comment by tabladrum** [comment]
I am good. @benathmane please rebase. Thanks

**Comment by tabladrum** [comment]
LGTM

**Comment by amitmawkin** [comment]
@benathmane also please update the READM.md with the following (Community Contribution) besides the title

**Comment by amitmawkin** [comment]
LGTM

**Comment by benathmane** [comment]
ERROR into : Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.29:npm (npm install) on project UI: Failed to run task: 'npm install' failed. (error code 1) -> [Help 1] !!!!!!!!!!!!!!!!

**Comment by benathmane** [comment]
How to fix this problem ?

File: collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/GitlabSettings.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    public void setApiToken(String apiToken) {
+        this.apiToken = apiToken;
+    }
+
+    public String getBranche() {
+        return branche;
+    }
+
+    public void setBranche(String branche) {
```
Comment (MarkRx) [inline collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/GitlabSettings.java]: typo?

**Comment by tabladrum** [comment]
@benathmane all's good. Please rebase and we will merge. Thanks!!

**Comment by benathmane** [comment]
@tabladrum please rebuild travis and merge :) Thanks.

**Comment by tabladrum** [comment]
Rebuilding... Not sure it failed build again.

**Comment by benathmane** [comment]
@tabladrum please merge :) Thanks.

**Comment by tabladrum** [comment]
Nice! Thanks @benathmane for the contrib.

**Pull request merged**